src/transactions/test/ClaimableBalanceTests.cpp

```diff
@@ -0,0 +1,1029 @@
+
+            // authorize trustline
+            if (!isNative)
+            {
+                SECTION("no trust")
+                {
+                    REQUIRE_THROWS_AS(acc1.createClaimableBalance(
+                                          asset, amount, validClaimants),
+                                      ex_CREATE_CLAIMABLE_BALANCE_NO_TRUST);
+                }
+            }
+
+            // create and modify trustlines regardless of asset type so the
+            // native balance changes by the same amount in both scenarios
+            // create unauthorized trustline
+            issuer.setOptions(
+                setFlags(static_cast<uint32_t>(AUTH_REQUIRED_FLAG)));
+            acc1.changeTrust(usd, trustLineLimit);
+
+            if (!isNative)
+            {
+                SECTION("not authorized")
+                {
+                    REQUIRE_THROWS_AS(
+                        acc1.createClaimableBalance(asset, amount,
+                                                    validClaimants),
+                        ex_CREATE_CLAIMABLE_BALANCE_NOT_AUTHORIZED);
+                }
+            }
+
+            issuer.allowTrust(usd, acc1);
+
+            SECTION("underfunded")
+            {
+                REQUIRE_THROWS_AS(
+                    acc1.createClaimableBalance(asset, amount, validClaimants),
+                    ex_CREATE_CLAIMABLE_BALANCE_UNDERFUNDED);
+            }
+
+            auto fundForClaimableBalance = [&]() {
+                // provide enough balance to acc1 to pay amount on the claimable
+                // balance
+                if (!isNative)
+                {
+                    issuer.pay(acc1, usd, amount);
+                }
+                else
+                {
+                    root.pay(acc1, amount);
+                }
+            };
+
+            SECTION("low reserve")
+            {
+                // Include second claimant. An additional baseReserve is
+                // required, but will not be available. Account was created with
+                // 3 baseReserves, but we need 4 (1 for account, 1 for
+                // trustline, and 2 for claimable balance entry)
+
+                validClaimants.emplace_back(makeClaimant(issuer, simplePred));
+                REQUIRE_THROWS_AS(
+                    acc1.createClaimableBalance(asset, amount, validClaimants),
+                    ex_CREATE_CLAIMABLE_BALANCE_LOW_RESERVE);
+
+                fundForClaimableBalance();
+
+                // add reserve and try again
+                root.pay(acc1, minBalance1);
+                acc1.createClaimableBalance(asset, amount, validClaimants);
+            }
+
+            fundForClaimableBalance();
+
+            SECTION("valid")
+            {
+                Claimant c;
+                c.v0().destination = acc2;
+
+                for (uint32_t i = 0; i < 4; i++)
+                {
+                    SECTION("predicate at level " + std::to_string(i))
```

Comment:
from recent findings, let's be defensive and avoid using `to_string` -> use `fmt::format` instead

Reply:
done